src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -239,22 +264,51 @@ private static void updateAuthorList(RepoConfiguration config) throws NoAuthorsW
         }
     }
 
-    private static void generateIndividualRepoReport(
-            CommitContributionSummary commitSummary, AuthorshipSummary authorshipSummary, String repoReportDirectory) {
-        CommitReportJson commitReportJson = new CommitReportJson(commitSummary, authorshipSummary);
-        FileUtil.writeJsonFile(commitReportJson, getIndividualCommitsPath(repoReportDirectory));
-        FileUtil.writeJsonFile(authorshipSummary.getFileResults(), getIndividualAuthorshipPath(repoReportDirectory));
+    /**
+     * Adds {@code failedRepoConfig} into the collated list of failed repos and logs into the list of errors in the
+     * summary.
+     */
+    private static void handleBranchingFailed(RepoConfiguration failedRepoConfig) {
+        ErrorSummary.getInstance().addErrorMessage(failedRepoConfig.getDisplayName(),
+                String.format(LOG_BRANCH_DOES_NOT_EXIST, failedRepoConfig.getBranch()));
+        failedRepoConfigsList.add(failedRepoConfig);
+    }
+
+    /**
+     * Adds all {@code failedRepoConfigs} into the list of failed repos and logs into the list of errors in the
+     * summary.
+     */
+    private static void handleCloningFailed(List<RepoConfiguration> failedRepoConfigs) {
+        for (RepoConfiguration failedConfig : failedRepoConfigs) {
+            ErrorSummary.getInstance().addErrorMessage(failedConfig.getDisplayName(),
+                    String.format(LOG_ERROR_CLONING, failedConfig.getLocation()));
+            failedRepoConfigsList.add(failedConfig);
+        }
+    }
+
+    /**
+     * Removes configs from {@code repoConfigs} that failed to clone or branch out previously.
+     */
+    private static void removeFailedRepoConfigs(List<RepoConfiguration> repoConfigs) {
+        repoConfigs.removeAll(failedRepoConfigsList);
```

Comment:
instead of this, perhaps we can remove it on the go, using an iterator?
https://stackoverflow.com/questions/1196586/calling-remove-in-foreach-loop-in-java

Reply:
Considering this approach. Meanwhile, is there a reason to use an iterator over the current approach?

Reply:
it is unsafe to remove elements within a for each loop, hence iterator

Reply:
Let's use `Collection.removeIf` instead, as the underlying function implement iterator to remove the elements as well.

Reply:
The rationale is to remove the method, removeFailedRepoConfigs as a whole by removing elements within the for each loop you're using itself. Reason -> we try to avoid global variables whenever possible.

Reply:
Noted. I've removed the method `removeFailedRepoConfigs` and remove the failed configs on the go.